pkg/apis/compliance/v1alpha1/tailoredprofile_types.go

```diff
@@ -6,6 +6,21 @@ import (
 
 // FIXME: move name/rationale to a common struct with an interface?
 
+// DisableOutdatedReferenceValidation a label is used to disable validation of outdated references
+const DisableOutdatedReferenceValidation = "compliance.openshift.io/outdated-reference-validation-disable"
+
+// PurneOutdatedReferencesAnnotationKey is the annotation key used to indicate that the outdated references of rules or variables should be pruned
+const PurneOutdatedReferencesAnnotationKey = "compliance.openshift.io/prune-outdated-references"
+
+// RuleLastCheckTypeChangedAnnotationKey is the annotation key used to indicate that the rule check type has changed, store its previous check type
+const RuleLastCheckTypeChangedAnnotationKey = "compliance.openshift.io/rule-last-check-type"
+
+// VariableDeprecatedAnnotationKey is the annotation key used to indicate that the variable is deprecated
+const VariableDeprecatedAnnotationKey = "compliance.openshift.io/variable-deprecated"
+
+// DeprecatedRuleAnnotationKey is the annotation key used to indicate that the rule is deprecated
+const DeprecatedRuleAnnotationKey = "compliance.openshift.io/deprecated-rule"
+
```

Comment:
It is not clear to me what is assigning the deprecated annotations to the rules. I guess they would be added by the profile parser based on some content processing?

If they are not being assigned right now, maybe their handling should be assigned later?
It would be nice to have tests for these too.

Reply:
Let's create this later in a separate patch, creating a Jira task for it

Reply:
Okay